src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -608,6 +610,18 @@ abstract class AbstractCheckTask(
             null
         }
 
+    private fun TaskTimeout.getOrCalculateMessageTimeout(): Long {
+        if (this.timeout > 0L) {
+            return timeout
+        }
+        require(messageTimeout != null && messageTimeout > 0) {
+            "Timeout cannot be calculated because 'timeout' and 'message timeout' is not set"
+        }
+        return messageTimeout.also {
```

Comment:
I don't think that we can use the `messageTimeout` as a parameter for the timer task to stop our rule. The timer task is executed in real-time. So the `timeout` must be greater that the `messageTimeout`

Reply:
So we should to add an additional timeout to the message timeout? Something like this `message timeout + 5000 mills`